wgpu-core/src/command/transfer.rs

```diff
@@ -40,6 +43,49 @@ pub struct TextureCopyView {
     pub origin: Origin3d,
 }
 
+/// Error encountered while attempting a data transfer.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum TransferError {
+    /// The source buffer/texture is missing the `COPY_SRC` usage flag.
+    MissingCopySrcUsageFlag,
+    /// The destination buffer/texture is missing the `COPY_DST` usage flag.
+    MissingCopyDstUsageFlag,
+    /// Some dimension of the buffer/image is not properly aligned.
+    DimensionNotMultipleOf {
+        /// Name of the problematic buffer/image dimension.
+        dimension: &'static str,
```

Comment:
Do you think we could replace this `DimensionNotMultipleOf` with these:
- UnalignedCopySize
- UnalignedBufferOffset
- UnalignedCopyWidth
- UnalignedCopyHeight
- UnalignedBytesPerRow
- UnalignedRowsPerImage

We don't have to provide all the information about what went wrong, what alignment is expected, etc. These aren't supposed to be recoverable errors. Nobody should take these values and try to re-encode the copy with different parameters.

So it's enough if the description said, for example "`Xxx::bytes_per_row` is not aligned". They can look at the method documentation and see what the expectations are.